extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/health/RedisHealthCheck.java

```diff
@@ -17,41 +18,87 @@
 import org.eclipse.microprofile.health.Readiness;
 
 import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
 import io.quarkus.redis.client.RedisClient;
+import io.quarkus.redis.client.RedisClientName;
+import io.quarkus.redis.client.reactive.ReactiveRedisClient;
+import io.quarkus.redis.client.runtime.RedisClientUtil;
+import io.quarkus.redis.client.runtime.RedisConfig;
+import io.quarkus.redis.client.runtime.RedisConfig.RedisConfiguration;
 import io.vertx.redis.client.Response;
 
 @Readiness
 @ApplicationScoped
 class RedisHealthCheck implements HealthCheck {
-    private Map<String, RedisClient> clients = new HashMap<>();
+    private final Map<String, RedisClient> clients = new HashMap<>();
+    private final Map<String, ReactiveRedisClient> reactiveClients = new HashMap<>();
+    private final RedisConfig redisConfig;
+
+    public RedisHealthCheck(RedisConfig redisConfig) {
+        this.redisConfig = redisConfig;
+    }
 
     @PostConstruct
     protected void init() {
-        Set<Bean<?>> beans = Arc.container().beanManager().getBeans(RedisClient.class);
-        for (Bean<?> bean : beans) {
-            if (bean.getName() == null) {
-                // this is the default redis client: retrieve it by type
-                RedisClient defaultClient = Arc.container().instance(RedisClient.class).get();
-                clients.put(DEFAULT_CLIENT, defaultClient);
-            } else {
-                RedisClient client = (RedisClient) Arc.container().instance(bean.getName()).get();
-                clients.put(bean.getName(), client);
+        for (InstanceHandle<RedisClient> handle : Arc.container().select(RedisClient.class, Any.Literal.INSTANCE).handles()) {
+            String clientName = getClientName(handle.getBean());
+            clients.put(clientName == null ? DEFAULT_CLIENT : clientName, handle.get());
+        }
+
+        for (InstanceHandle<ReactiveRedisClient> handle : Arc.container()
+                .select(ReactiveRedisClient.class, Any.Literal.INSTANCE).handles()) {
+            String clientName = getClientName(handle.getBean());
+            reactiveClients.put(clientName == null ? DEFAULT_CLIENT : clientName, handle.get());
```

Comment:
So we're not exactly on par with the MongoDB impl here where the name of the default reactive client is different?

Reply:
My reason was to product healthcheck status for each individual redis client without caring much whether that was done with a reactive client or a blocking one (as it does not matter much).

Reply:
I think we should be consistent but that can probably be done in another PR.

Merging.